packages/core/react/src/WalletProvider.tsx

```diff
@@ -240,10 +247,10 @@ export const WalletProvider: FC<WalletProviderProps> = ({
     );
 
     // Sign a transaction if the wallet supports it
-    const signTransaction = useMemo(
+    const signTransaction: SignerWalletAdapterProps['signTransaction'] | undefined = useMemo(
```

Comment:
This type is still _perfect_ in the sense that a dev will see `(transaction: Transaction | VersionedTransaction)` as the parameter type, so they have to know to check the adapter for `supportedTransactionVersions`. But it is functional and nonbreaking, because it will always be expected to return whichever type it was called with.

Reply:
That seems totally fine to me